supporter-product-data/src/main/scala/com/gu/model/zuora/request/BatchQueryRequest.scala

```diff
@@ -0,0 +1,32 @@
+package com.gu.model.zuora.request
+
+import io.circe.{Encoder, Json}
+import io.circe.generic.semiauto.deriveEncoder
+
+case class ZoqlExportQuery(
+  name: String,
+  query: String
+)
+
+case class BatchQueryRequest(
+  partner: String, // Makes the request stateful. See https://knowledgecenter.zuora.com/Central_Platform/API/AB_Aggregate_Query_API/BA_Stateless_and_Stateful_Modes
+  name: String,
+  queries: List[ZoqlExportQuery]
+)
+
+object ZoqlExportQuery{
+  implicit val encoder: Encoder[ZoqlExportQuery] = deriveEncoder[ZoqlExportQuery].mapJsonObject(_
+    .add("type", Json.fromString("zoqlexport"))
+  )
+}
+
+object BatchQueryRequest{
+  implicit val encoder: Encoder[BatchQueryRequest] = deriveEncoder[BatchQueryRequest].mapJsonObject(_
+    .add("format", Json.fromString("csv"))
```

Comment:
just a thought - the data compresses excellently (but not the query execution time) if you specify gzip here, do you think it's worth it?  There is also content-encoding header which would make more sense as it's more transparent but I'm not sure if zuora respects that.

Reply:
I did think about it and decided against it. The size of the file is not really a problem even for a full query so it doesn't seem worth the extra complexity